www/utils/createPages.js

```diff
@@ -39,39 +44,56 @@ async function createPages(graphql, actions, reporter) {
     reporter.panicOnBuild('🚨  ERROR: Loading createPages query');
   }
   // Create component detail pages.
-  const components = result.data.allMdx.edges;
+  const pages = result.data.allMdx.edges;
 
   const themesSCSSVariables = await getThemesSCSSVariables();
 
   // you'll call `createPage` for each result
-  // eslint-disable-next-line no-restricted-syntax
-  for (const { node } of components) {
-    const componentDir = node.slug.split('/')[0];
-    const variablesPath = path.resolve(__dirname, `../../src/${componentDir}/_variables.scss`);
-    const githubEditPath = `https://github.com/openedx/paragon/edit/master/src${node.fileAbsolutePath.split('src')[1]}`;
-    let scssVariablesData = {};
+  for (const { node } of pages) {
+    const githubEditPath = `https://github.com/openedx/paragon/edit/master/src${node.internal.contentFilePath.split('src')[1]}`;
+
+    if (node.fields.source === 'components') {
+      // Check for a _variables.scss file for this component, e.g. src/Button/_variables.scss.
+      // If it exists, load the data:
+      let scssVariablesData = {};
+      const componentDir = path.dirname(node.internal.contentFilePath);
+      const variablesPath = componentDir + '/_variables.scss';
```

Comment:
nit: there is a GitHub annotation for this line around "Unexpected string concatenation". Should it be addressed (i.e., using a template literal instead)? Also curious why CI isn't failing if there's a check failure annotation?

Reply:
Weird. It clearly failed in the CI but the "Test > Lint" check overall is still marked as passing:

![Screenshot](https://github.com/user-attachments/assets/693e446e-4419-426e-aa5b-bd82665e9b74)

Could be related to https://github.com/openedx/paragon/issues/3329 ?

Anyhow, I addressed the lint issue and squashed it into the relevant existing commit.